packages/fx-core/src/core/middleware/envInfoLoader.ts

```diff
@@ -200,6 +211,24 @@ async function askTargetEnvironment(
   }
 }
 
+async function useUserSetEnv(
+  ctx: CoreHookContext,
+  inputs: Inputs,
+  allowCreateNewEnv: boolean
+): Promise<string | undefined> {
+  const checkEnv = await environmentManager.checkEnvExist(inputs.projectPath!, inputs.env);
+  if (checkEnv.isErr()) {
+    ctx.result = checkEnv.error;
+    return undefined;
+  }
+  if (checkEnv.value || allowCreateNewEnv) {
+    return inputs.env;
```

Comment:
we might need to validate the user input env name here, as it might contain invalid characters and cannot be saved into a env profile file.

Reply:
Will add later.